ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessor.kt

```diff
@@ -0,0 +1,50 @@
+package com.revenuecat.purchases.ui.revenuecatui.data.processed
+
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.ui.revenuecatui.helpers.Logger
+import java.util.Locale
+
+internal object VariableProcessor {
+    private val REGEX = Regex("\\{\\{\\s[a-zA-Z0-9_]+\\s\\}\\}")
+
+    fun processVariables(
+        variableDataProvider: VariableDataProvider,
+        originalString: String,
+        rcPackage: Package,
+        locale: Locale,
+    ): String {
+        var resultString = originalString
+        REGEX.findAll(originalString).toList().reversed().forEach { matchResult ->
+            val newValue = variableValue(variableDataProvider, matchResult, rcPackage, locale)
+            newValue?.let {
+                resultString = resultString.replaceRange(matchResult.range, it)
+            }
+        }
+        return resultString
+    }
+
+    private fun variableValue(
+        variableDataProvider: VariableDataProvider,
+        matchResult: MatchResult,
+        rcPackage: Package,
+        locale: Locale,
+    ): String? {
+        val variableString = matchResult.value
+        return when (val variableWithoutBraces = variableString.substring(2, variableString.length - 2).trim()) {
```

Comment:
It's weird that this method does this instead of taking the variable name `String` directly. Would this be better in `processVariables`?